### DISCUSSION TITLE
Ip address for static pages via middleware

### DISCUSSION THREAD

**Discussion by SC0d3r** [vercel/next.js#39810] [Help]
**Hi everyone** 👋 I want to get the user ip for my static pages, and the only way to do that (as far as I know) is by using the `middleware` otherwise cause I am using `getStaticProps` I cant get the `req object`, but now the issue is that the req object in the middleware is different than the req object in the `getServerSideProps` , in serverSideProps I cant get the user ip but in the middleware the headers are not set for that.
the documentation in the nextjs site says that your host should provide the ip or geo field in the middleware req object, but I am self hosting my app, and how does a host add those fields to the req? 
if thats not possible then I have to use the getServerSideProps for my static pages just to get the user ip and make those pages **`slow!`**
Ps: I have read all the questions/asnwers on this subject in `stackoverflow` and issues in `vercel/next.js` but non of those answer my question.

What I have read so far:
[How does NextRequest gets the ip from the request? #33435](https://github.com/vercel/next.js/discussions/33435)
[how to get the ip address of the client from server side in next.js app?](https://stackoverflow.com/questions/68338838/how-to-get-the-ip-address-of-the-client-from-server-side-in-next-js-app)
[Next JS how to Redirect user based on the IP address?](https://stackoverflow.com/questions/59159852/next-js-how-to-redirect-user-based-on-the-ip-address)
[Get the client IP on NextJS and use SSR](https://stackoverflow.com/questions/66111742/get-the-client-ip-on-nextjs-and-use-ssr)
[Next.js api route with IP address](https://stackoverflow.com/questions/68506345/next-js-api-route-with-ip-address)
[How to use Nextjs middleware to redirect users to sub-domain based on IP](https://stackoverflow.com/questions/73408526/how-to-use-nextjs-middleware-to-redirect-users-to-sub-domain-based-on-ip)
[NextJS Middleware](https://nextjs.org/docs/advanced-features/middleware)
[next/server](https://nextjs.org/docs/api-reference/next/server)
[NextJS Headers](https://nextjs.org/docs/api-reference/next.config.js/headers)

any thoughts on this would be appreciated 🙏

**Comment by kznkv** [comment]
Hi, maybe someone has a solution how to get an IP?

**Comment by wjm0913** [comment]
> **Hi everyone** 👋 I want to get the user ip for my static pages, and the only way to do that (as far as I know) is by using the `middleware` otherwise cause I am using `getStaticProps` I cant get the `req object`, but now the issue is that the req object in the middleware is different than the req object in the `getServerSideProps` , in serverSideProps I cant get the user ip but in the middleware the headers are not set for that. the documentation in the nextjs site says that your host should provide the ip or geo field in the middleware req object, but I am self hosting my app, and how does a host add those fields to the req? if thats not possible then I have to use the getServerSideProps for my static pages just to get the user ip and make those pages **`slow!`** Ps: I have read all the questions/asnwers on this subject in `stackoverflow` and issues in `vercel/next.js` but non of those answer my question.
> 
> What I have read so far: [How does NextRequest gets the ip from the request? #33435](https://github.com/vercel/next.js/discussions/33435) [how to get the ip address of the client from server side in next.js app?](https://stackoverflow.com/questions/68338838/how-to-get-the-ip-address-of-the-client-from-server-side-in-next-js-app) [Next JS how to Redirect user based on the IP address?](https://stackoverflow.com/questions/59159852/next-js-how-to-redirect-user-based-on-the-ip-address) [Get the client IP on NextJS and use SSR](https://stackoverflow.com/questions/66111742/get-the-client-ip-on-nextjs-and-use-ssr) [Next.js api route with IP address](https://stackoverflow.com/questions/68506345/next-js-api-route-with-ip-address) [How to use Nextjs middleware to redirect users to sub-domain based on IP](https://stackoverflow.com/questions/73408526/how-to-use-nextjs-middleware-to-redirect-users-to-sub-domain-based-on-ip) [NextJS Middleware](https://nextjs.org/docs/advanced-features/middleware) [next/server](https://nextjs.org/docs/api-reference/next/server) [NextJS Headers](https://nextjs.org/docs/api-reference/next.config.js/headers)
> 
> any thoughts on this would be appreciated 🙏

I have encountered the same problem as you, do you have a solution?
